asset-transfer-private-data/chaincode-java/src/main/java/org/hyperledger/fabric/samples/privatedata/AssetTransfer.java

```diff
@@ -0,0 +1,579 @@
+
+        //Make submitting client the owner
+        asset.setOwner(clientID);
+        System.out.printf("CreateAsset Put: collection %s, ID %s\n", ASSET_COLLECTION_NAME, assetID);
+        System.out.printf("  Put: collection %s, ID %s\n", ASSET_COLLECTION_NAME, new String(asset.serialize()));
+        stub.putPrivateData(ASSET_COLLECTION_NAME, assetID, asset.serialize());
+
+        // Get collection name for this organization.
+        String orgCollectionName = getCollectionName(ctx);
+
+        //Save AssetPrivateDetails to org collection
+        AssetPrivateDetails assetPriv = new AssetPrivateDetails(assetID, appraisedValue);
+        System.out.printf("  Put AssetPrivateDetails: collection %s, ID %s\n", orgCollectionName, assetID);
+        stub.putPrivateData(orgCollectionName, assetID, assetPriv.serialize());
+
+        return asset;
+    }
+
+    /**
+     * AgreeToTransfer is used by the potential buyer of the asset to agree to the
+     * asset value. The agreed to appraisal value is stored in the buying orgs
+     * org specifc collection, while the the buyer client ID is stored in the asset collection
+     * using a composite key
+     * Uses transient map with key asset_value
+     *
+     * @param ctx the transaction context
+     */
+    @Transaction(intent = Transaction.TYPE.SUBMIT)
+    public void AgreeToTransfer(final Context ctx) {
+        ChaincodeStub stub = ctx.getStub();
+        Map<String, byte[]> transientMap = ctx.getStub().getTransient();
+        if (!transientMap.containsKey("asset_value")) {
+            String errorMessage = String.format("AgreeToTransfer call must specify \"asset_value\" in Transient map input");
+            System.err.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.INCOMPLETE_INPUT.toString());
+        }
+
+        byte[] transientAssetJSON = transientMap.get("asset_value");
+        AssetPrivateDetails assetPriv;
+        String assetID;
+        try {
+            JSONObject json = new JSONObject(new String(transientAssetJSON, UTF_8));
+            assetID = json.getString("assetID");
+            final int appraisedValue = json.getInt("appraisedValue");
+
+            assetPriv = new AssetPrivateDetails(assetID, appraisedValue);
+        } catch (Exception err) {
+            String errorMessage = String.format("TransientMap deserialized error %s ", err);
+            System.err.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.INCOMPLETE_INPUT.toString());
+        }
+
+        if (assetID.equals("")) {
+            String errorMessage = String.format("Invalid input in Transient map: " + "assetID");
+            System.err.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.INCOMPLETE_INPUT.toString());
+        }
+        if (assetPriv.getAppraisedValue() <= 0) { // appraisedValue field must be a positive integer
```

Comment:
If appraised value wasn't passed, would it be 0?

Reply:
verified that if appraisedVal is not passed, it goes into catch block in ln 367